### PR TITLE
[Merged by Bors] - chore(*): drop some long-deprecated theorems

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Option/Basic.lean [hunk 1]
```diff
@@ -319,6 +319,10 @@ theorem orElse_none' (x : Option α) : x.orElse (fun _ ↦ none) = x := by cases
 
 #align option.ne_none_iff_is_some Option.ne_none_iff_isSome
 
+@[simp]
```
Comment (grunweg) [inline Mathlib/Data/Option/Basic.lean:322]: Flagging this for fellow reviewers to note: this lemma is new (but very similar to exist ones).

**Review by grunweg — COMMENTED** [review]
Thanks for doing this!
maintainer merge

File: Mathlib/Data/Option/Basic.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Data/Option/Basic.lean:322]: Let me revert this `@[simp]` attr and move it to a new PR.
```suggestion
```

**Comment by grunweg** [comment]
The last two commits also look good to me - with one question: `AddMonoidHom.map_list_sum` was only deprecated recently; should it stay for some more time? (If I remember correctly, it was used in mathlib until May - so it's not an obviously bad lemma that can be removed without second thought.)

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**